scrapy/utils/curl.py

```diff
@@ -90,5 +90,9 @@ def curl_to_request_kwargs(curl_command, ignore_unknown_options=True):
         result['cookies'] = cookies
     if parsed_args.data:
         result['body'] = parsed_args.data
+        if result.get('method') == 'GET':
+            # if the "data" is specified but the "method" is not specified,
+            # the default method is 'POST'
+            result['method'] = 'POST'
```

Comment:
While it should not be a problem in most real-life scenarios, `curl <URL> -X GET --data asdf` works. We need to distinguish when `-X` is passed with `GET` and when it is omitted.

Reply:
Hi @Gallaecio, 

I'm not sure which approach to take. From my perspective, this command tries to do the best from the cURL command, and I think that it doesn't have any sense to add the body when the method is GET.

Before deciding this I checked other tools as https://curl.trillworks.com/ , and when entering:
```bash
curl URL -X GET --data asdf
```

I got:
```python
import requests

data = 'asdf'

response = requests.post('http://URL', data=data)
```

Of course, I can change it to respect the provided GET, but I'm not sure about the best option. Any other idea / thoughts?

Reply:
I’m not against merging the code as is.

But a server may support a GET with a form even if it is not standard:

```
[adrian@afonsox ~]$ curl httpbin.org/anything -X GET --data asdf
{
  "args": {}, 
  "data": "", 
  "files": {}, 
  "form": {
    "asdf": ""
  }, 
  "headers": {
    "Accept": "*/*", 
    "Content-Length": "4", 
    "Content-Type": "application/x-www-form-urlencoded", 
    "Host": "httpbin.org", 
    "User-Agent": "curl/7.70.0", 
    "X-Amzn-Trace-Id": "Root=…"
  }, 
  "json": null, 
  "method": "GET", 
  "origin": "…", 
  "url": "http://httpbin.org/anything"
}
```

So I would either implement `GET` support here or create a separate issue to eventually add it.

Reply:
Ok, I fixed the code to respect this behavior. :+1: